dnn-tuning-nvidia-mxnet/cnn_text/dataset.py

```diff
@@ -0,0 +1,43 @@
+"""
+Loads and preprocessed data for the MR dataset.
+Returns input vectors, labels, vocabulary, and inverse vocabulary.
+"""
+from collections import namedtuple
+
+import numpy as np
+
+from preprocess import (load_data_and_labels, pad_sentences,
+                        build_vocab, build_input_data)
+
+Data = namedtuple('Data', ['x_train', 'y_train', 'x_dev',
+                           'y_dev', 'vocab_size', 'sentence_size'])
+
+def get_data():
+    # Load and preprocess data
+    sentences, labels = load_data_and_labels()
+    sentences_padded = pad_sentences(sentences)
+    vocabulary, vocabulary_inv = build_vocab(sentences_padded)
+    x, y = build_input_data(sentences_padded, labels, vocabulary)
+
+    vocab_size = len(vocabulary)
+
+    # randomly shuffle data
+    np.random.seed(10)
```

Comment:
can you do this through a sklearn library? http://scikit-learn.org/stable/modules/cross_validation.html#random-permutations-cross-validation-a-k-a-shuffle-split Will this work?